client/consensus/babe/src/verification.rs

```diff
@@ -156,9 +157,17 @@ fn check_primary_header<B: BlockT + Sized>(
 ) -> Result<(), Error<B>> {
 	let author = &epoch.authorities[pre_digest.authority_index as usize].0;
 
+	let mut epoch_index = epoch.epoch_index;
+
+	if epoch.end_slot() <= pre_digest.slot {
+		// Slot doesn't strictly belong to this epoch, create a clone with fixed values.
+		let fixed_epoch = epoch.clone_for_slot(pre_digest.slot);
+		epoch_index = fixed_epoch.epoch_index;
+	}
```

Comment:
nit: mut could be avoided:
`let epoch_index = if epoch.end_slot() <= pre_digest.slot { ... } else {epoch.epoch_index}`